docs/software/devtools/debug/linaro.md

```diff
@@ -0,0 +1,294 @@
+
+Once configured, test and save the configuration:
+
+1. check whether the configuration is correct, click `Test Remote Launch`.
+2. Click on `ok` and `close` to save the configuration.
+3. You can now connect by going to `Remote Launch` and choose the `Alps` entry.
+   If the client fails to connect, look at the message, check your ssh
+   configuration and make sure you can ssh without the client.
+
+### Setup the user environment and build the executable
+
+Once the uenv is loaded and activated, the program to debug must be compiled
+with the `-g` (for cpu) and `-G` (for gpu) debugging flags. For example, we can
+build a cuda test with a user environment:
+
+```bash
+> uenv start prgenv-gnu:24.11:v1
+> uenv view default
+> nvcc -c -arch=sm_90 -g -G test_gpu.cu
+> mpicxx -g test_cpu.cpp test_gpu.o -o myexe
+```
+
+### Launch Linaro DDT
+
+To use the DDT client with uenv, it must be launched in `Manual Launch` mode
+(assuming that it is connected to Alps via `Remote Launch`):
+
+=== "on local machine"
+
+    Start DDT, and connect to the target cluster using the drop down menu for Remote Launch.
+
+    Click on Manual launch, set the number of processes to listen to, 
+    then wait for the slurm job to start (see the "on Alps" tab).
+        
+    <img src="https://raw.githubusercontent.com/jgphpc/cornerstone-octree/ddt/scripts/img/ddt/0.png" width="600" />
+
+=== "on Alps"
+
+    log into the system and launch with the srun command:
+
+    ```bash
+    # start a session with both the PE used to build your application
+    # and the linaro-forge uenv mounted
+    > uenv start prgenv-gnu/24.11:v1,linaro-forge/24.1.1:v1 --view=prgenv-gnu:default
+    > source /user-tools/activate
+
+    > srun -N1 -n4 -t15 -pdebug \
+        ./cuda_visible_devices.sh   ddt-client   ./myexe
+    ```
+
+### Start debugging
+
+By default, DDT will pause execution on the call to MPI_Init:
+<img src="https://raw.githubusercontent.com/jgphpc/cornerstone-octree/ddt/scripts/img/ddt/1.png" width="600" />
+
+There are more than 1 mechanism for controlling program execution:
+
+=== "Breakpoint"
+
+    Breakpoint(s) can be set by clicking in the margin to the left of the line number:
+
+    <img src="https://raw.githubusercontent.com/jgphpc/cornerstone-octree/ddt/scripts/img/ddt/3.png" width="600" />
+
+=== "Stop at"
+
+    Execution can be paused in every CUDA kernel launch by activating the default breakpoints from the Control menu:
+
+    <img src="https://raw.githubusercontent.com/jgphpc/cornerstone-octree/ddt/scripts/img/ddt/4.png" width="400" />
+
+
+This screenshot shows a debugging session on 128 gpus:
+![DDTgpus](https://raw.githubusercontent.com/jgphpc/cornerstone-octree/ddt/scripts/img/ddt/5.png)
+
+More informations regarding how to use Linaro DDT are provided in the Forge
+[User Guide](https://docs.linaroforge.com/latest/html/forge/index.html).
+
+## Troubleshooting
+
+Notes on using specific systems:
+
+=== "alps"
```

Comment:
Since there is only one system, is this really necessary? I'd also prefer subsections here, so they are easily reachable from the navigation menu.